src/client/telemetry/index.ts

```diff
@@ -1032,14 +1046,29 @@ export interface IEventNamePropertyMapping {
     [Telemetry.RawKernelStartRawSession]: never | undefined;
     [Telemetry.RawKernelProcessLaunch]: never | undefined;
 
+    // Applies to everything (interactive+Notebooks & local+remote)
+    [Telemetry.NotebookStart]:
+        | ResourceSpecificTelemetryProperties // If successful.
+        | (ResourceSpecificTelemetryProperties & TelemetryErrorProperties); // If there any any unhandled exceptions.
```

Comment:
The benefit of sending `TelemetryErrorProperties` with this same event is that we know whether this event failed or not. 
Previously we'd send a seprate event `ERROR` and that would contain a property `originalEvent` & then we'd need to find out whether an event failed. (a very convoluted approach).
Now we can just query based on property `failed = true`